### PR TITLE
policy: Add runtime tests for ToFQDN rules

### PR DESCRIPTION
**Pull request by raybejjani**
Test ToFQDN in a simple manner, checking that it allows the IPs
intended, and no others.

Fixes 92243feb5df4956e87874b94ba77278d67ba3206

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/4908)

### CODE REVIEW

**Comment by raybejjani** [comment]
test-me-please

**Review by ianvernon — APPROVED** [review]

**Comment by raybejjani** [comment]
test-me-please

File: test/runtime/Policies.go [hunk 1]
```diff
@@ -874,6 +874,54 @@ var _ = Describe("RuntimeValidatedPolicies", func() {
...
+		By("Denying egress to IPs of non-ToFQDN DNS names that are looked up")
+		for _, blockedTarget := range []string{"www.cilium.io", "google.com", "yahoo.com", "1.1.1.1", "127.0.0.1"} {
+			res := vm.ContainerExec(helpers.App1, helpers.CurlFail(blockedTarget))
+			res.ExpectFail("Curl succeeded against blocked DNS name " + blockedTarget)
+		}
+
+		By("Allowing egress to IPs of specified ToFQDN DNS names")
+		allowedTarget := "cilium.io"
+		res := vm.ContainerExec(helpers.App1, helpers.CurlWithHTTPCode(allowedTarget))
+		res.ExpectContains("301", fmt.Sprintf("Cannot access "+allowedTarget+"%v", res.OutputPrettyPrint()))
```
Comment (eloycoto) [inline test/runtime/Policies.go]: you don't need to use fmt.Sprintf, I would change this line to: 
```
res.ExpectContains("301", "Cannot access to %q err: %s", allowedTarget, res.OutputPrettyPrint())
```

File: test/runtime/Policies.go [hunk 2]
```diff
@@ -874,6 +874,54 @@ var _ = Describe("RuntimeValidatedPolicies", func() {
...
+    ]
+  }
+]`
+		_, err := vm.PolicyRenderAndImport(fqdnPolicy)
+		Expect(err).To(BeNil(), "Unable to import policy: %s", err)
+
+		By("Testing that ToFQDN enables policy enforcement")
+
+		By("Denying egress to IPs of non-ToFQDN DNS names that are looked up")
+		for _, blockedTarget := range []string{"www.cilium.io", "google.com", "yahoo.com", "1.1.1.1", "127.0.0.1"} {
```
Comment (eloycoto) [inline test/runtime/Policies.go]: Are all these targets needed? Each one will take Curl Max timeout (3 seconds) so not sure that we need to test all. 

On the other hand, I would add here another test, something like `cilium.io.customdomain.io` that looks like as another corner case.

File: test/runtime/Policies.go (hunk 1 above)
Comment (raybejjani) [reply test/runtime/Policies.go]: Oh, excellent. I'll change it! thanks

File: test/runtime/Policies.go (hunk 2 above)
Comment (raybejjani) [reply test/runtime/Policies.go]: you're right. They're actually pretty redundant. I'll use `www.cilium.io` (exists but should block) and `cilium.io.cilium.io` (doesn't exist and should block, and it tests if we somehow break if the domain repeats). I need an actual IP in there, though, just as a sanity check. 

Including your suggestion, there are 3 negative tests, each taking 3s. Is that ok?

**Comment by raybejjani** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Comment by raybejjani** [comment]
test-me-please

**Comment by raybejjani** [comment]
test-me-please

**Comment by raybejjani** [comment]
test-me-please
